crates/pathfinder/src/state/merkle_tree.rs

```diff
@@ -463,6 +522,27 @@ impl<T: NodeStorage> MerkleTree<T> {
         Ok(result)
     }
 
+    /// Generates a proof for the value located at `key`, as described in [#714](https://github.com/eqlabs/pathfinder/issues/714).
+    ///
+    /// Returns a vector of `ProofNode` which can then be verified by the verifier
+    /// (provided the verifier knows `root`, `key` and `value`).
+    pub fn get_proof(&self, key: &BitSlice<Msb0, u8>) -> anyhow::Result<Vec<ProofNode>> {
+        let mut nodes = self.traverse(key)?;
+
+        // Return an error if the list is empty
+        let node = nodes.last().context("Tree is empty")?;
```

Comment:
Took the decision to return an error if the list is empty. Returning `Ok([])` would make less sense to me because, asking a proof on an empty tree will*necessarily*  be `invalid` from my pov.

Reply:
I guess it's a philosophiscal question at this point haha :p I'm ok with returning `[]`: I just think from a user perspective, I'd prefer the node to throw me "You are requesting for something stupid because the tree is empty" rather than "hey ok, here's your proof: []".

Both work for me: implemented in 0e3659eea49b1ddc322f428c32825e3b70390386 :)